server/authorization.go

```diff
@@ -313,12 +313,10 @@ func (a *AuthorizationServer) createEnvoyForwarded(
 
 	// apigee dynamic data response headers
 	var dynamicDataHeaders []*corev3.HeaderValueOption
+	// envRequest being nil means envoy adapter is standing alone and we don't need to send those headers.
 	if envRequest != nil {
-		var apiSpec *config.APISpec
-		if spec := envRequest.GetAPISpec(); spec != nil {
-			apiSpec = spec
-		}
-		dynamicDataHeaders = apigeeDynamicDataHeaders(a.handler.Organization(), a.handler.Environment(), api, apiSpec, nil)
+		msgID := req.GetAttributes().GetRequest().GetHttp().GetHeaders()[headerMessageID]
```

Comment:
Should this still create an ID if the header isn't present?

Reply:
Makes sense! Made the change inside the func.